Documentation/ceph-backup.md

```diff
@@ -0,0 +1,483 @@
+    machineDisruptionBudgetNamespace: openshift-machine-api
+```
+
+### Create an Object Store to confirm that data is restored
+
+We are going to create an Object Store, create a bucket and add some items. So, on restore we can confirm that all data is restored.
+
+Here are the commands. However for a deeper tutorial go to [Ceph Object Storage documentation](ceph-object.md).
+
+The Bucket Claim definition is as follows:
+
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: ceph-delete-bucket
+  namespace: rook-ceph
+spec:
+  # To create a new bucket specify either `bucketName` or 
+  # `generateBucketName` here. Both cannot be used. To access
+  # an existing bucket the bucket name needs to be defined in
+  # the StorageClass referenced here, and both `bucketName` and
+  # `generateBucketName` must be omitted in the OBC.
+  #bucketName: 
+  bucketName: ceph-bkt
+  storageClassName: rook-ceph-delete-bucket
+  additionalConfig:
+    # To set for quota for OBC
+    #maxObjects: "1000"
+    #maxSize: "2G"
+```
+
+Execute the following commands in order:
+
+```console
+# Create Object Store
+kubectl apply -f object.yaml
+
+# Bucket Creation
+kubectl apply -f storageclass-bucket-delete.yaml
+kubectl apply -f object-bucket-claim-delete.yaml
+
+# Connect to Bucket, execute at background in new terminal
+kubectl port-forward -n rook-ceph service/rook-ceph-rgw-my-store 8080:80
+
+# Add data
+export AWS_HOST=`kubectl get configmap -n rook-ceph ceph-delete-bucket -o 'jsonpath={.data.BUCKET_HOST}';echo`
+export AWS_ENDPOINT=localhost:8080
+export BUCKET_NAME=`kubectl get configmap -n rook-ceph ceph-delete-bucket -o 'jsonpath={.data.BUCKET_NAME}';echo`
+export AWS_ACCESS_KEY_ID=`kubectl get secret -n rook-ceph ceph-delete-bucket -o 'jsonpath={.data.AWS_ACCESS_KEY_ID}' | base64 --decode;echo`
+export AWS_SECRET_ACCESS_KEY=`kubectl get secret -n rook-ceph ceph-delete-bucket -o 'jsonpath={.data.AWS_SECRET_ACCESS_KEY}' | base64 --decode;echo`
+
+for i in {1..10};do aws s3 cp /etc/hosts s3://${BUCKET_NAME}/$i --endpoint-url http://${AWS_ENDPOINT};done
+
+# List data
+aws s3 ls s3://${BUCKET_NAME} --endpoint-url http://${AWS_ENDPOINT}
+```
+
+## Install and configure Velero
+
+[Velero](https://velero.io/) is an open source tool to safely backup and restore, perform disaster recovery, and migrate Kubernetes cluster resources and persistent volumes.
+
+For backing up and restoring Kubernetes volumes you need to use the [restic plugin](https://velero.io/docs/v1.5/restic/). You need to add `--use-restic` to velero install command.
+
+Velero supports a lot of cloud providers. Take at look in the [supported providers](https://velero.io/docs/v1.5/supported-providers/) section of the doc for more information and examples. 
+
+For example for [GCP](https://github.com/vmware-tanzu/velero-plugin-for-gcp#setup).
+
+```console
+
+# Configure Google Storage bucket and service account.
+BUCKET=<YOUR_BUCKET>
```

Comment:
Is there Velero documentation that we could reference? We don't want to maintain documentation in Rook for Velero install instructions.

Reply:
Yes, the link above this code is the official documentation for Velero in GCP.